### PR TITLE
Update online.lua

### PR DESCRIPTION
**Pull request by Colandus**
Shorter, cleaner, but mainly solves the issue where ghost players are counted. Has been thoroughly tested with different amounts of players online. No problem in splitting or whatsoever.

### CODE REVIEW

File: data/talkactions/scripts/online.lua [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
 	end
 
-	if i > 0 then
-		msg = msg .. "."
-		player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, msg)
+	local playersOnline = #onlineList
+	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, playersOnline .. " players online.")
+	
+	local i = 1
+	while i <= playersOnline do
```
Comment (Mkalo) [inline data/talkactions/scripts/online.lua]: `for i = 1, playersOnline, maxPlayersPerMessage do`

File: data/talkactions/scripts/online.lua [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
-		if i == 10 then
-			if k == playerCount then
-				msg = msg .. "."
-			else
-				msg = msg .. ","
-			end
-			player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, msg)
-			msg = ""
-			i = 0
+			table.insert(onlineList, targetPlayer:getName() .. " [" .. targetPlayer:getLevel() .. "]")
```
Comment (Mkalo) [inline data/talkactions/scripts/online.lua]: string.format("%s [%d]", targetPlayer:getName(), targetPlayer:getLevel())

File: data/talkactions/scripts/online.lua (hunk 1 above)
Comment (Colandus) [inline data/talkactions/scripts/online.lua]: Thanks. No idea why I didn't do that.

File: data/talkactions/scripts/online.lua [hunk 3]
```diff
@@ -1,36 +1,22 @@
...
 	local hasAccess = player:getGroup():getAccess()
 	local players = Game.getPlayers()
-	local playerCount = Game.getPlayerCount()
-
-	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, playerCount .. " players online.")
+	local onlineList = {}
 
-	local i = 0
-	local msg = ""
 	for k, targetPlayer in ipairs(players) do
```
Comment (ranisalt) [inline data/talkactions/scripts/online.lua]: You don't need the value of k anymore, use `for _, targetPlayer in ipairs(players)`

File: data/talkactions/scripts/online.lua [hunk 4]
```diff
@@ -1,36 +1,22 @@
...
-		if i == 10 then
-			if k == playerCount then
-				msg = msg .. "."
-			else
-				msg = msg .. ","
-			end
-			player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, msg)
-			msg = ""
-			i = 0
+			table.insert(onlineList, ("%s [%d]"):format(targetPlayer:getName(), targetPlayer:getLevel()))
```
Comment (ranisalt) [inline data/talkactions/scripts/online.lua]: Why do you use string `:format(...)` here but use concatenation below?

File: data/talkactions/scripts/online.lua (hunk 3 above)
Comment (Colandus) [inline data/talkactions/scripts/online.lua]: same

File: data/talkactions/scripts/online.lua (hunk 4 above)
Comment (Colandus) [inline data/talkactions/scripts/online.lua]: I added format per Mkalo's request. I'll add that too.

**Comment by Colandus** [comment]
What do you guys think about allowing an option to sort by level or name or even grouping by vocation? As a config.lua setting. Or is this too much?

File: data/talkactions/scripts/online.lua (hunk 3 above)
Comment (ranisalt) [inline data/talkactions/scripts/online.lua]: Sure, but _ usually has the semantics of "I don't care what is this value", at least in Lua and Python 😄

**Comment by ranisalt** [comment]
> Or is this too much?

Yes. Sorting is up to the server manager.

File: data/talkactions/scripts/online.lua [hunk 5]
```diff
@@ -1,36 +1,22 @@
...
 
-	if i > 0 then
-		msg = msg .. "."
-		player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, msg)
+	local playersOnline = #onlineList
+	player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, ("%d players online."):format(playersOnline))
+	
+	for i = 1, playersOnline, maxPlayersPerMessage do
+		local msg = table.concat(onlineList, ", ", i, math.min(i + maxPlayersPerMessage - 1, playersOnline))
+		player:sendTextMessage(MESSAGE_STATUS_CONSOLE_BLUE, msg .. ".")
```
Comment (ranisalt) [inline data/talkactions/scripts/online.lua]: Move the `.. "."` to the line above where the message is being built.

Comment (ranisalt) [inline data/talkactions/scripts/online.lua]: Move the `.. "."` to the line above where `msg` is being build.

**Comment by Colandus** [comment]
@ranisalt alrighty, is this fine?

**Comment by ranisalt** [comment]
Thanks!

**Pull request merged**